From e4b136ce9753b3d52ea1bb89920c2bc305c24a7c Mon Sep 17 00:00:00 2001
From: Matt Wagantall <mwagantall@cyngn.com>
Date: Tue, 24 Nov 2015 12:11:11 -0800
Subject: [PATCH 5/5] legacy: perf: allow checking for existence of other
 processes (signull)

Certain perfd optimizations depend on this ability.

Change-Id: Ib994cf879db73c02d8c79c8b7e6a8a855496c6be

sepolicy: perfd: fix signull permission

Commit 1a20a7fbc2 ("sepolicy: perf: allow checking for existence of
other processes (signull)") was implemented incorrectly. It granted
perfd only permissions to signull itself, not other processes. Fix
this, granting signull permissions to all processes by using the
'domain' attribute.

Change-Id: I5ea7c543ba5854550bb020382b43368d75244f32
---
 legacy-common/perfd.te | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/legacy-common/perfd.te b/legacy-common/perfd.te
index 0d2bed1..67b0583 100644
--- a/legacy-common/perfd.te
+++ b/legacy-common/perfd.te
@@ -47,6 +47,9 @@ allow perfd mpctl_data_file:file create_file_perms;
 # Thermal lib access
 unix_socket_connect(perfd, thermal, thermal-engine)
 
+# Allow perfd to check for existence of other processes
+allow perfd domain:process signull;
+
 # Allow perfd to set properties
 set_prop(perfd, freq_prop)
 
-- 
2.17.0

